### PR TITLE
FT2023-342: Adding the files for the routing task.

### PR DESCRIPTION
**Pull request by Diwakar-59**

### CODE REVIEW

File: web/modules/custom/helloworld/helloworld.info.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: Hello world
+type: module
+description: My custom Hello World module for Drupal 10.
+package: Custom Module
+version: 1.0
+core_version_requirement: ^10
+configure: helloworld.admin_settings
+dependencies:
+  - drupal:views
+  - drupal:media
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/helloworld.info.yml]: This red circle at the end of file means the expected new line at the end of the file is missing.
Please add the new line at the end of the file for all files showing this symbol.
Please configure VS code to automatically add this.

File: web/modules/custom/helloworld/helloworld.module [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?php 
+
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+
+/**
+ * @file
+ * Implements hooks.
+ */
+
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/helloworld.module:7]: unnecessary white space.

File: web/modules/custom/helloworld/src/Controller/HelloController.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<?php 
+
+namespace Drupal\helloworld\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/src/Controller/HelloController.php]: unnecessary extra whitespace.

File: web/modules/custom/helloworld/src/Controller/HelloController.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * Class for handling the routes of helloworld module.
+ */
+class HelloController extends ControllerBase {
+    /**
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/src/Controller/HelloController.php]: Empty line should be there after class name line.

File: web/modules/custom/helloworld/src/Controller/HelloController.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+   * Display the markup.
+   *
+   * @return array
+   *   Return markup array.
+   */
+  public function content() {
+    $user = \Drupal::currentUser();
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/src/Controller/HelloController.php]: Using services like this in classes is not acceptable.
Please use dependency injection instead.

File: web/modules/custom/helloworld/src/Form/HelloworldSettingsForm.php [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    return [
+      'helloworld.admin_settings',
+    ];
+  }
+
+  /**
+   * Function responsible for building the form elements.
+   *
+   * @param array $form
+   * @param FormStateInterface $form_state
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/src/Form/HelloworldSettingsForm.php]: Please check any core files to know how function docblock is written properly.
Please ensure the same is fixed for all other docblocks in this PR.

File: web/modules/custom/helloworld/src/Form/HelloworldSettingsForm.php [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    if(!preg_match(
+      '/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/', 
+      $form_state->getValue('email'))) {
+        $form_state->setErrorByName(
+          'email', $this->t('Enter a valid email address!!'));
+      }
+  }
+
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/src/Form/HelloworldSettingsForm.php:138]: Unnecessary extra whitespace.

File: web/modules/custom/helloworld/src/Form/HelloworldSettingsForm.php [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+   * @param array $form
+   * @param FormStateInterface $form_state
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $this->config('helloworld.admin_settings')
+    ->set('full_name', $form_state->getValue('full_name'))
+    ->set('email', $form_state->getValue('email'))
+    ->set('phone_number', $form_state->getValue('phone_number'))
+    ->set('gender', $form_state->getValue('gender'))
+
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/src/Form/HelloworldSettingsForm.php]: unnecessary extra newline

File: web/modules/custom/routing_module/src/Controller/RoutingController.php [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+class RoutingController extends ControllerBase {
+
+  /**
+   * Function to render the content page based on the route.
+   *
+   * @param mixed $slug
+   * @return array
+   */
+  public function content($slug = null) {
+    $user = \Drupal::currentUser();
```
Comment (vivekpan27) [inline web/modules/custom/routing_module/src/Controller/RoutingController.php]: Please use dependency injection instead.

**Review by vivekpan27 — CHANGES_REQUESTED** [review]
Please fix the feedbacks.

**Comment by Diwakar-59** [comment]
> Please fix the feedbacks.

Updated the files accordingly.

File: web/modules/custom/helloworld/helloworld.info.yml (hunk 1 above)
Comment (vivekpan27) [inline web/modules/custom/helloworld/helloworld.info.yml]: @Diwakar-59 Why do I still see dependencies which are not required?

File: web/modules/custom/helloworld/helloworld.module [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+/**
+ * Implements hook form alter.
+ *
+ * @param array $form
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The FormStateInterface object.
+ * @param string $form_id
+ * @return void
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/helloworld.module]: This is not required when implementing hooks.
Please check some examples from core.

File: web/modules/custom/helloworld/helloworld.module [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+ */
+function helloworld_form_alter(&$form, \Drupal\Core\Form\FormStateInterface $form_state, $form_id) {
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/helloworld.module]: We don't mention namespace in the function arguments.
Please check how it is done in Core.

File: web/modules/custom/helloworld/src/Controller/HelloController.php [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Session\AccountInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+  /**
+   * Class for handling the routes of helloworld module.
+   */
+class HelloController extends ControllerBase {
+
+  protected $account;
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/src/Controller/HelloController.php:20]: Where is the docblock for this?

File: web/modules/custom/helloworld/src/Controller/HelloController.php [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('current_user')
+    );
+  }
+
+  /**
+   * Display the markup.
+   *
+   * @return array
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/src/Controller/HelloController.php:44]: You also need to mention @throws here.
Please check for examples from core.

File: web/modules/custom/helloworld/src/Form/HelloworldSettingsForm.php [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+use Drupal\Core\Form\FormStateInterface;
+
+  /**
+   * Class to handle the configuration form of the helloworld custom module.
+   */
+  class HelloworldSettingsForm extends ConfigFormBase {
+
+  /**
+   * Function to get the form id of the configure form.
+   * @return string
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/src/Form/HelloworldSettingsForm.php]: There should be an empty line in between.

File: web/modules/custom/helloworld/src/Form/HelloworldSettingsForm.php (hunk 6 above)
Comment (vivekpan27) [reply web/modules/custom/helloworld/src/Form/HelloworldSettingsForm.php]: This is still not fixed.

File: web/modules/custom/helloworld/src/Form/HelloworldSettingsForm.php [hunk 15]
```diff
@@ -0,0 +1,150 @@
...
+    if(!preg_match(
+      '/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/', 
+      $form_state->getValue('email'))) {
+        $form_state->setErrorByName(
+          'email', $this->t('Enter a valid email address!!'));
+      }
+  }
+
+  /**
+   * Fuction to handle the submission of the form and updating the fields data.
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/src/Form/HelloworldSettingsForm.php]: Typo

File: web/modules/custom/helloworld/src/Form/HelloworldSettingsForm.php [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+   *
+   * @param array $form
```
Comment (vivekpan27) [inline web/modules/custom/helloworld/src/Form/HelloworldSettingsForm.php]: This is not the correct way of writing the docblock.

**Review by vivekpan27 — CHANGES_REQUESTED** [review]
Still a lot of fixes to be made.

**Pull request merged**